framework/kms-keyring.md

```diff
@@ -309,7 +312,84 @@ If the response is successfully verified, OnDecrypt MUST do the following with t
 If OnDecrypt fails to successfully decrypt any [encrypted data key](#structures.md#encrypted-data-key),
 then OnDecrypt MUST output the unmodified input [decryption materials](#structures.md#decryption-materials).
 
+## Configuration Intent
+
+### OnEncrypt Goal
+
+When a user configures a KMS keyring with key IDs
+and use that keyring to encrypt a message,
+they are stating their intent that they need each one of those CMKs to be able to
+independently decrypt the resulting encrypted message.
+
+For example, if a user configures a KMS keyring with CMKs A, B, and C
+and uses that keyring to encrypt a message,
+then that user expects three things to be true:
+
+1. CMK A can decrypt the encrypted message.
+1. CMK B can decrypt the encrypted message.
+1. CMK C can decrypt the encrypted message.
+
+If any of these are not true of the resulting encrypted message,
+the keyring has failed to honor the user's intent.
+
+In order to accomplish this,
+the KMS keyring MUST contribute three encrypted data keys to the encryption materials:
+one from CMK A, one from CMK B, and one from CMK C.
+
+### OnDecrypt Goal
+
+When a user configures a KMS keyring for use on decrypt,
+they are stating their intent for which CMKs
+the keyring will *attempt* to use to decrypt encrypted data keys.
+
+For example, if a user configures a KMS keyring with CMK C
+and uses it to decrypt an encrypted message
+that contains encrypted data keys for CMKs A, B, and C,
+then the keyring will attempt to decrypt using CMK C.
+
+However, if the keyring attempts to decrypt using CMK C and cannot,
+this failure still honors the configured intent and MUST NOT halt decryption.
+The configured intent is that they keyring MUST *attempt* with these CMKs,
```

Comment:
I would try to make it a bit more clear that `these CMKs` refers to the CMKs configured with this keyring, not "CMK A, B, and C". Maybe make this more clear by explicitly giving the example of "If you configure the KMS keyring for decrypt with CMKs C, D, E" or something like that? Or is that more confusing?